src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs

```diff
@@ -19,34 +23,183 @@ public partial class SplashScreen : Window
-            webBrowserUserDataFolder = userDataDir.Exists ? userDataDir : null;
-
             webView = new WebView2();
             AddChild(webView);
+            // Bind event handlers
+            webView.NavigationCompleted += WebView_NavigationCompleted;
+            DynamoModel.RequestUpdateLoadBarStatus += DynamoModel_RequestUpdateLoadBarStatus;
+            StaticSplashScreenReady += FinishLoadingDynamicScreen;
+            RequestLaunchDynamo = LaunchDynamo;
+            RequestImportSettings = ImportSettings;
+            RequestSignIn = SignIn;
+            RequestSignOut = SignOut;
+        }
+
+        private void WebView_NavigationCompleted(object sender, CoreWebView2NavigationCompletedEventArgs e)
+        {
+            SetLabels();
+            if (webView != null)
+            {
+                webView.NavigationCompleted -= WebView_NavigationCompleted;
+            }
+        }
+
+        /// <summary>
+        /// Import setting file from chosen path
+        /// </summary>
+        /// <param name="fileContent"></param>
+        private void ImportSettings(string fileContent)
+        {
+            bool isImported = viewModel.PreferencesViewModel.importSettingsContent(fileContent);
+            if (isImported)
+            {
+                SetImportStatus(ImportStatus.success);
+            }
+            else
+            {
+                SetImportStatus(ImportStatus.error);
+            }
+            Analytics.TrackEvent(Actions.ImportSettings, Categories.SplashScreenOperations, isImported.ToString());
+        }
+
+        /// <summary>
+        /// Returns true if the user was successfully logged in, else false.
+        /// </summary>
+        private bool SignIn()
+        {
+            authManager.Login();
+            bool ret = authManager.IsLoggedIn();
+            Analytics.TrackEvent(Actions.SignIn, Categories.SplashScreenOperations, ret.ToString());
+            return ret;
+        }
+
+        /// <summary>
+        /// Returns true if the user was successfully logged out, else false.
+        /// </summary>
+        /// <returns></returns>
+        private bool SignOut()
+        {
+            authManager.Logout();
+            bool ret = !authManager.IsLoggedIn();
+            Analytics.TrackEvent(Actions.SignOut, Categories.SplashScreenOperations, ret.ToString());
+            return ret;
+        }
+
+        /// <summary>
+        /// Handler to launch Dynamo View
+        /// </summary>
+        /// <param name="isCheckboxChecked"></param>
+        private void LaunchDynamo(bool isCheckboxChecked)
+        {
+            viewModel.PreferenceSettings.EnableStaticSplashScreen = !isCheckboxChecked;
+            DynamoModel.RequestUpdateLoadBarStatus -= DynamoModel_RequestUpdateLoadBarStatus;
+            StaticSplashScreenReady -= FinishLoadingDynamicScreen;
+            Close();
+            Application.Current.MainWindow = dynamoView;
```

Comment:
I guess this line will change if we plan to show the SplashScreen from Revit when Dynamo is starting.

Reply:
This line will not change, just the variable of `dynamoView` will point to the view initialized in DynamoRevit